lib/web_ui/lib/src/engine/view_embedder/dimensions_provider/custom_element_dimensions_provider.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/src/engine/window.dart';
+import 'package:ui/ui.dart' as ui show Size;
+
+import '../../dom.dart';
+import 'dimensions_provider.dart';
+
+// import 'custom_element_application_dom.dart';
+// import 'full_page_application_dom.dart';
+// import 'hot_restart_cache_handler.dart';
+
+/// This class provides the real-time dimensions of a "hostElement".
+///
+/// Note: all the measurements returned from this class are potentially
+/// *expensive*, and should be cached as needed. Every call to every method on
```

Comment:
A pattern we use for expensive methods is use prefix `compute` instead of `get`.

Do you think we can remove the need for these by replacing with `ResizeObserver` and such?

Reply:
Right now we're only listening to the "resize" events to THEN go and `compute` the physical size of the host element (this is inherited from the visualViewport approach).

ResizeObserver, however, returns the dimensions rect of the observer, which we could cache and use the next time that computePhysicalSize happens. Even when there's no resize, we get an immediate event with the size of the element that we're observing as the observer connects.

So for custom elements: yes, we could use only a ResizeObserver and cache the last metric we've seen for each observed element as its "physicalSize".

I suspect that for the `visualViewport`, we can't use this pattern, because the VisualViewport object is an EventTarget (and not an HtmlElement).